File: core/src/main/java/org/dreamexposure/discal/core/object/web/WebPartialGuild.java
package org.dreamexposure.discal.core.object.web;

public class WebPartialGuild {
    private final long id;
    private final String name;
    private final String iconUrl;

    public WebPartialGuild(final long id, final String name, final String iconUrl) {
        this.id = id;
        this.name = name;
        this.iconUrl = iconUrl;
    }

    //Getters
    public long getId() {
        return this.id;
    }

    public String getName() {
        return this.name;
    }

    public String getIconUrl() {
        return this.iconUrl;
    }
}

File: core/src/main/kotlin/org/dreamexposure/discal/core/object/web/WebPartialGuild.kt
package org.dreamexposure.discal.core.`object`.web

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.builtins.LongAsStringSerializer

@Serializable
data class WebPartialGuild(
        @Serializable(with = LongAsStringSerializer::class)
        val id: Long,
        val name: String,
        @SerialName("icon_url")
        val iconUrl: String,
)
